File: Core/Models/Vehicle.cs
using System.Collections.ObjectModel;

namespace Vega.Core.Models
{
    public class Vehicle
    {
        public int Id { get; set; }
        public int ModelId { get; set; }
        public Model? Model {get; set;}
        public bool IsRegistered { get; set; }
        public Contact? Contact {get; set;}
        public DateTime LastUpdate { get; set; }
        public ICollection<Feature> Features { get; set; }

        public Vehicle()
        {
            Features = new Collection<Feature>();
        }
    }
}

File: ClientApp/src/app/models/vehicle.ts
export interface IdNamePair {
    id: number;
    name: string;
}

export interface Contact {
    name: string;
    phone: string;
    email: string;
}

export interface Vehicle {
    id: number;
    model: IdNamePair;
    make: IdNamePair;
    isRegistered: boolean;
    features: IdNamePair[];
    contact: Contact;
    lastUpdate: string;
}

export interface SaveVehicle {
    id: number;
    modelId?: number;
    makeId: number;
    isRegistered: boolean;
    features: number[];
    contact: Contact;
}